tck/src/test/java/io/smallrye/context/tck/lifecycle/LifecycleExecuter.java

```diff
@@ -54,11 +59,12 @@ private void unregisterJTA() throws Exception {
         namingBean.stop();
     }
 
-    /*
-     * public void executeBeforeDeploy(@Observes BeforeDeploy event, TestClass testClass) {
-     * execute(testClass.getMethods(io.smallrye.context.tck.lifecycle.api.BeforeDeploy.class));
```

Comment:
@FroMage do you happen to know why we had this in place? I don't even know why we have [`BeforeDeploy`](https://github.com/smallrye/smallrye-context-propagation/blob/master/tck/src/test/java/io/smallrye/context/tck/lifecycle/api/BeforeDeploy.java) class under tck, it seems useless...

Reply:
Let me be clear about CDI once and for all: I've no clue how any of it works, so I trust people like @stuartwdouglas and you to figure it out by yourselves ;)

Reply:
If I merged this too quickly, just let me know what I need to do to fix things. Sorry I should have waited for your review.

Reply:
No, the CDI part is perfectly fine (I was just about to approve the PR anyway).
But this bit has nothing to do with CDI, it's Arq. extension stuff.

Reply:
It was commented out, I replaced the commented out stuff with a real BeforeDeploy